tap_recharge/discover.py

```diff
@@ -1,20 +1,46 @@
-from singer.catalog import Catalog, CatalogEntry, Schema
-from tap_recharge.schema import get_schemas, STREAMS
+from singer.catalog import Catalog
+from tap_recharge.schema import get_schemas
+
+def _get_key_properties_from_meta(schema_meta: list) -> str:
+    """
+    Gets the table-key-properties from the schema metadata.
+    """
+    return schema_meta[0].get('metadata').get('table-key-properties')
+
+def _get_replication_method_from_meta(schema_meta: list) -> str:
+    """
+    Gets the forced-replication-method from the schema metadata.
+    """
+    return schema_meta[0].get('metadata').get('forced-replication-method')
+
+def _get_replication_key_from_meta(schema_meta: list) -> str:
+    """
+    Gets the valid-replication-keys from the schema metadata.
+    """
+    if _get_replication_method_from_meta(schema_meta) == 'INCREMENTAL':
+        return schema_meta[0].get('metadata').get('valid-replication-keys')[0]
+    return None
 
 def discover():
+    """
+    Constructs a singer Catalog object based on the schemas and metadata.
+    """
     schemas, field_metadata = get_schemas()
-    catalog = Catalog([])
+    streams = []
+
+    for schema_name, schema in schemas.items():
+        schema_meta = field_metadata[schema_name]
 
-    for stream_name, schema_dict in schemas.items():
-        schema = Schema.from_dict(schema_dict)
-        mdata = field_metadata[stream_name]
+        catalog_entry = {
+            'stream': schema_name,
+            'tap_stream_id': schema_name,
+            'schema': schema,
+            'key_properties': _get_key_properties_from_meta(schema_meta),
+            'replication_method': _get_replication_method_from_meta(schema_meta),
+            'replication_key': _get_replication_key_from_meta(schema_meta),
```

Comment:
These keys are not needed at this level of the catalog entry. Does the tap use them?

https://github.com/singer-io/getting-started/blob/master/docs/DISCOVERY_MODE.md#the-catalog

Reply:
I believe this is how the cookiecutter template I've been using sets up `discover.py` and the catalog, but the tap doesn't use this so I will remove it and the associated functions.